src/components/app/UrlManager.js

```diff
@@ -200,17 +200,16 @@ class UrlManagerImpl extends React.PureComponent<Props> {
       return;
     }
 
+    const oldUrl = window.location.pathname + window.location.search;
     const newUrl = urlFromState(urlState);
-    if (newUrl !== window.location.pathname + window.location.search) {
-      if (!getIsHistoryReplaceState()) {
-        // Push the URL state only when the url setup is done, and we haven't set
-        // a flag to only replace the state.
-        window.history.pushState(urlState, document.title, newUrl);
-      } else {
-        // Replace the URL state before the URL setup is done, and if we've specifically
-        // flagged to replace the URL state.
-        window.history.replaceState(urlState, document.title, newUrl);
-      }
+    if (!window.history.state || getIsHistoryReplaceState()) {
+      // Replace the URL state if we've specifically flagged to replace the URL
+      // state. This happens during the loading process. Also do it if we don't
+      // have an history state yet.
+      window.history.replaceState(urlState, document.title, newUrl);
+    } else if (newUrl !== oldUrl) {
+      // Push the URL state when we haven't set a flag to only replace the state.
+      window.history.pushState(urlState, document.title, newUrl);
```

Comment:
This is doing 2 things:
1. when `getIsReplaceState` is true, we always call `replaceState`. Previously we were doing it only if the URL was changing, but sometimes the URL doesn't change when the `urlState` changes. For example these problematic maps `hiddenLocalTracksByPid` and `localTrackOrderByPid` needs to have all the entries for all pids, but the serialized URL won't change if some entries have empty Set.
2. Also if we don't have an `history.state` we add it too.

Reply:
Interesting, I think it makes sense to replace the state with the new one if its's `isHistoryReplaceState`. 

Sometimes I feel like our url state handling is pretty broken and we should rewrite it 😅 I think I contributed a lot to this messiness when I first joined to this project as well 😅 .